### PR TITLE
adding date field choise on instance and instance periods (draft, proposal)

### PR DESCRIPTION
**Pull request by jackjack82**
The goal is to have a report based on different date field on the move line.
The date field in on instance periods, in order to enable comparison between accounting/cash flow date in different columns.

The date field is also available on the report instance with "apply" button, in order to apply the date directly to all periods.

image

Improvements/open points:
- rename new field (e.g. date_field_name)?
- what if the date field is empty in destination model? Set a "undefined" line/field or warning?
- remove date field selction from non-"actuals" columns
- update automatic tests
- test flow adding new object to the MIS builder (other than account.move.line)

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Isn't better to create an SQL view from account_move_line with the date field you want and use that source? I think this complicates things without a real benefit.

**Comment by jackjack82** [comment]
This was a request from a client and there is an open issue https://github.com/OCA/mis-builder/issues/91 . We may add additional dates on move lines, e.g. payment date. 

This can also be partially solved using existing queries. If not interested you can close the PR and the issue. It was a proposal, we have no final reports or case use yet.

Thank you, 
Giacomo

**Comment by pedrobaeza** [comment]
Let's wait for @sbidoul's opinion. He's on holidays, so this might delay a bit, but if there's no hurry, it's OK to wait.

**Comment by sbidoul** [comment]
@jackjack82 thanks for this PR!

My first reaction is this adds quite a bit of complexity for something that can be done today with a SQL view such as the one shown in https://github.com/OCA/mis-builder/issues/91#issuecomment-387402890 (albeit with some development).

However, I'm curious about your customer's use case. Maybe by collecting those use case such as #91 an interesting solution will emerge.

**Comment by jjscarafia** [comment]
thanks @jackjack82 this is quite near to what we where looking for on https://github.com/OCA/mis-builder/issues/91.

I understand the caution of @sbidoul and @pedrobaeza . For me one use case would be for cash flow exactly as shown on https://github.com/OCA/mis-builder/issues/91. I believe is not much complexity to be added but I agree that another use case would be needed to think on a change like this.

Regarding the possibility of an SQL view from account_move_line, it's a good idea but obviously it's much easier just to be able to make a report based on the due_date. I believe being able to use due_date on mis reports would extend possibilities, not just for cash flow Forecast.

In summary, I agree with @sbidoul on collecting more use cases
